algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -458,6 +458,59 @@ case class CMSItem[K](item: K, override val totalCount: Long, override val param
 }
 
 /**
+ * A sparse Count-Min sketch structure, used for situations where the key is highly skewed.
+ */
+case class SparseCMS[K](exactCountTable: Map[K, Long],
+  override val totalCount: Long,
+  override val params: CMSParams[K]) extends CMS[K](params) {
+
+  override def +(x: K, count: Long): CMS[K] = {
+    val currentCount = exactCountTable.getOrElse(x, 0L)
+    val newTable = exactCountTable.updated(x, currentCount + count)
+    if (newTable.size < width * depth) {
+      // still sparse
+      SparseCMS(newTable, totalCount + count, params)
+    } else {
+      // Create new CMSInstace
+      var cms = CMSInstance[K](params)
+      newTable.foreach {
```

Comment:
just to be consistent, let's use foldLeft rather than foreach here (to avoid an unneeded var). I'm willing to have vars if they add performance, in this case the perf should be the same as:
```scala
newTable.foldLeft(CMSInstance[K](params))(_ + _)
```